pkg/blob/controllerserver.go

```diff
@@ -691,3 +807,58 @@ func parseDays(dayStr string) (int32, error) {
 
 	return int32(days), nil
 }
+
+func GenerateSASToken(accountName, accountKey string) (string, error) {
+	credential, err := azblob.NewSharedKeyCredential(accountName, accountKey)
+	if err != nil {
+		return "", status.Errorf(codes.Internal, fmt.Sprintf("failed to generate sas token in creating new shared key credential, accountName: %s, accountKey: %s, err: %s", accountName, accountKey, err.Error()))
+	}
+	serviceClient, err := service.NewClientWithSharedKeyCredential(fmt.Sprintf("https://%s.blob.core.windows.net/", accountName), credential, nil)
+	if err != nil {
+		return "", status.Errorf(codes.Internal, fmt.Sprintf("failed to generate sas token in creating new client with shared key credential, accountName: %s, accountKey: %s, err: %s", accountName, accountKey, err.Error()))
+	}
+	sasURL, err := serviceClient.GetSASURL(
+		sas.AccountResourceTypes{Object: true, Service: true, Container: true},
+		sas.AccountPermissions{Read: true, List: true, Write: true, Delete: false, Add: false, Create: false, Update: false},
```

Comment:
I think by default is false, so you only need to set `true`, pls provide least permissions in the sas url.

Reply:
done